health-services/project-factory/src/server/api/campaignApis.ts

```diff
@@ -793,8 +781,15 @@ async function handleResouceDetailsError(request: any, error: any) {
   if (request?.body?.Activities && Array.isArray(request?.body?.Activities) && request?.body?.Activities.length > 0) {
     logger.info("Waiting for 2 seconds");
     await new Promise(resolve => setTimeout(resolve, 2000));
-    const activityObject = request?.body?.Activities;
-    await produceModifiedMessages(activityObject, config?.kafka?.KAFKA_CREATE_RESOURCE_ACTIVITY_TOPIC);
+
+    const activities = request?.body?.Activities;
+    const chunkPromises = [];
+    for (let i = 0; i < activities.length; i += 50) {
```

Comment:
the batch size should come from application property, multiple catch block one for the kafka message size too big issue and other for rest all type of exception